tripy/tripy/flat_ir/flat_ir.py

```diff
@@ -113,16 +113,18 @@ def to_mlir_impl():
                         arg_attrs: List[Dict[str, ir.Attribute]] = []
                         for bound in self.shapes:
                             # TODO (#244): Support multiple profiles
-
-                            arg_attrs.append(
-                                {
-                                    "tensorrt.shape_profile": ir.Attribute.parse(
-                                        f"#tensorrt.shape_profile<min={list(bound.min)}, opt={list(bound.opt)}, max={list(bound.max)}>"
+                            if bound.is_static():
+                                arg_attrs.append(ir.DictAttr.get({}))
```

Comment:
Doesn't this need to be all of nothing? What if you have a static tensor between two dynamic tensors? How does MLIR-TRT know that the shape profiles should map to the first and third tensors?

Reply:
```
  func.func @main(%arg0: tensor<?xf32> {tensorrt.shape_profile = #tensorrt.shape_profile<min = [1], opt = [2], max = [5]>}, %arg1: tensor<1xf32>, %arg2: tensor<?xf32> {tensorrt.shape_profile = #tensorrt.shape_profile<min = [1], opt = [2], max = [5]>}) -> tensor<?xf32>
```
`arg_attrs` is a list of `ir.DictAttr` with the same length as #inputs. For inputs that don't require any attribute, you just set empty `DictAttr`.

Reply:
Ah got it, misread.